src/env.cc

```diff
@@ -364,7 +374,20 @@ bool Environment::RemovePromiseHook(promise_hook_func fn, void* arg) {
 void Environment::EnvPromiseHook(v8::PromiseHookType type,
                                  v8::Local<v8::Promise> promise,
                                  v8::Local<v8::Value> parent) {
-  Environment* env = Environment::GetCurrent(promise->CreationContext());
+  Local<v8::Context> context = promise->CreationContext();
+
+  // Grow the embedder data if necessary to make sure we are not out of bounds
+  // when reading the magic number.
+  context->SetAlignedPointerInEmbedderData(ContextEmbedderIndex::kContextTagBoundary, nullptr);
+  int *magicNumberPtr = reinterpret_cast<int *>(
+    context->GetAlignedPointerFromEmbedderData(ContextEmbedderIndex::kContextTag)
+  );
+  if (magicNumberPtr != kNodeContextTagPtr || *magicNumberPtr != kNodeContextTag) {
```

Comment:
there might be the reaaaaally tiny small chance that the pointer might actually point to the "random" memory location yet that this random location does not contain the expected value?

Reply:
That's assuming that the context is actually created by node, maybe there could be the chance that:
a) some other program (e.g. electron uses that field number)
b) that by a (very small) chance that other program stores on that field a pointer that happens to match the expected pointer
c) and that the pointer contents are the "magic number" (extremely unlikely)

I mean, it is a failsafe, but I can just remove it and set the magic number to 0 and just rely on the pointer address if you want.

Reply:
@xaviergonz Yeah, but if they manage to get the same pointer into the same internal field, then we still won’t be able to distinguish these. :)

> I mean, it is a failsafe, but I can just remove it and set the magic number to 0 and just rely on the pointer address if you want.

I think it’s good to use a magic value here so that it’s easier to recognize – just saying that the second condition here will currently never be true if the first one was false. :)

Reply:
done